src/app/providers/profile/profile.ts

```diff
@@ -155,6 +171,69 @@ export class ProfileProvider {
     }
   }
 
+  public setWalletGroupsHome(walletObj) {
+    let result = _.cloneDeep(this.resultPrimaryAccount);
+    let walletsGroupsHome = [];
+    walletsGroupsHome = JSON.parse(localStorage.getItem("listHome")) ? JSON.parse(localStorage.getItem("listHome")) : [];
+    if (walletObj && walletObj.walletId) {
+      if (walletsGroupsHome.length < 5) {
+        if (!JSON.stringify(walletsGroupsHome).includes(JSON.stringify(walletObj))) {
+          walletsGroupsHome.push(walletObj);
+          localStorage.setItem("listHome", JSON.stringify(walletsGroupsHome));
+          result.added.status = true;
+          return result;
+        }
+        result.duplicate.status = true;
+        return result;
+      }
+      result.full.status = true;
+      return result;
+    } 
+    return result;
+  }
+
+  public async getWalletGroupsHome() {
+    let walletsGroupsHome = [];
+    let data = JSON.parse(localStorage.getItem("listHome"));
+    if (data) {
+      walletsGroupsHome = data.map((item) => {
+        return this.getWalletPrimary(item.walletId, item.tokenId);
+      })
+    }
+    // Handle case delete Key
+    if (walletsGroupsHome.includes(undefined)) {
+      let tempWalletsGroupsHome = [];
+      walletsGroupsHome = _.compact(walletsGroupsHome);
+      tempWalletsGroupsHome = walletsGroupsHome.map((item) => {
+        return {
+          walletId: item?.id,
+          tokenId: item?.tokens?.tokenId
+        };
+      });
+      localStorage.setItem("listHome", JSON.stringify(tempWalletsGroupsHome));
+    }
+    return walletsGroupsHome;
+  }
+
+  public removeWalletGroupsHome(walletObj) {
+    let data = JSON.parse(localStorage.getItem("listHome"));
+    if (data) {
+      const isExist = data.find((item) => {
+        if (walletObj && walletObj.tokenId && walletObj.walletId) {
+          return item.tokenId === walletObj.tokenId;
+        } else if (walletObj && walletObj.walletId && !walletObj.tokenId) {
+          return item.walletId === walletObj.walletId && item.tokenId === walletObj.tokenId;
+        } else {
+          return undefined;
+        }
+      });
+      data.splice(data.indexOf(isExist), 1);
```

Comment:
isExist will return undefined if not found data => so data.indexOf(undefined) = -1 => data.splice(-1, 1);
it will splice data, but no isExist found . check this code

Reply:
Check again in case of isExist = 0 => first element of list
If(isExist) => in this case isExist = 0 means false will not jump in this if statement
Therefore, it will not splice for first data .
Suggest : use isExist > -1 ( indexOf will return -1 if not found )